### PR TITLE
Add VisionPublisher and Detection- and GeometrySupplier

### PR DESCRIPTION
**Pull request by thumbnail**
todo
 - [x] Add JavaDoc
 - [x] Merge and rebase on top of #28 

A proposed main could be as follows:
```java
  public static void main(String[] args) throws UnknownHostException {
    Flux.combineLatest(
      new SSLVisionPublisher(InetAddress.getByName("localhost"), 123),
      new MulticastPublisher<>(InetAddress.getByName("localhost"), 1234, new UnsafeFunction<>(SSL_Referee::parseFrom)),
      InputFrame::new)
    .transform(new GameDeducer<>())
    .subscribe(System.out::println);
  }

  @Value
  public static class InputFrame implements GeometrySupplier, DetectionSupplier, SSLRefereeSupplier {
    @Delegate
    VisionFrame visionFrame;
    SSL_Referee SSLReferee;
  }
```

### CODE REVIEW

File: src/main/java/io/leonis/subra/ipc/network/GamePublisher.java [hunk 1]
```diff
@@ -23,21 +21,19 @@
  * @author Rimon Oz
  */
 @AllArgsConstructor
-public final class GamePublisher implements Publisher<GameFrame> {
-  private final Publisher<WrapperPacket> visionPublisher;
-  private final Publisher<SSL_Referee> refboxPublisher;
+public final class GamePublisher<I extends WrapperPacketSupplier & SSLRefereeSupplier> implements Deducer<I, GameFrame> {
```
Comment (romni) [inline src/main/java/io/leonis/subra/ipc/network/GamePublisher.java]: should be called `GameDeducer`?

File: src/main/java/io/leonis/subra/ipc/serialization/protobuf/SSLRefboxDeducer.java [hunk 2]
```diff
@@ -19,14 +19,14 @@
...
+    implements Deducer<I, Referee> {
 
   @Override
-  public Publisher<Referee.Supplier> apply(
-      final Publisher<SSL_Referee> refboxPublisher) {
+  public Publisher<Referee> apply(final Publisher<I> refboxPublisher) {
     return Flux.from(refboxPublisher)
-        .map(packet -> () -> new Referee.State(
+        .map(SSLRefereeSupplier::getSSLReferee)
+        .map(packet -> new Referee.State(
```
Comment (romni) [inline src/main/java/io/leonis/subra/ipc/serialization/protobuf/SSLRefboxDeducer.java]: 👍

Comment (thumbnail) [reply src/main/java/io/leonis/subra/ipc/serialization/protobuf/SSLRefboxDeducer.java]: This should be reverted

**Comment by thumbnail** [comment]
Mobile doesn't allow for changing the title; e.g. this is still WIP, because
 - [ ] `GamePublisher` should be removed
 - [ ] `RefereeDeducer` should return a `Referee.Supplier`

File: src/main/java/io/leonis/subra/ipc/network/GameDeducer.java [hunk 3]
```diff
@@ -2,42 +2,40 @@
...
-        Flux.from(this.visionPublisher).transform(new SSLVisionDeducer()),
-        Flux.from(this.refboxPublisher).transform(new SSLRefboxDeducer()),
-        GameFrameWithoutGoals::new)
-      .transform(new ParallelDeducer<>(
+  public Publisher<GameFrame> apply(final Publisher<I> entryPublisher) {
+    return Flux.from(entryPublisher)
+        .transform(new ParallelDeducer<>(
+            new SSLVisionDeducer<>(),
+            new SSLRefboxDeducer<>(),
+            GameFrameWithoutGoals::new))
```
Comment (romni) [inline src/main/java/io/leonis/subra/ipc/network/GameDeducer.java:34]: on r51 you want to have 

```
    <J extends Player.SetSupplier & GoalDimension.Supplier & Field.Supplier & Ball.SetSupplier>
    GameFrameWithoutGoals(final J packet, final Referee.Supplier referee) {
```

in order to get rid of the explicit dependency on `VisionPacket`

**Review by romni — APPROVED** [review]

**Pull request merged**